src/prommis/leaching/leach_flowsheet_dynamic.py

```diff
@@ -182,12 +204,55 @@ def set_inputs(m):
     m.fs.leach.mscontactor.solid[0, :].mass_frac_comp["Fe2O3"].fix()
     m.fs.leach.mscontactor.solid[0, :].flow_mass.fix()
 
-    m.fs.leach.mscontactor.liquid_inherent_reaction_extent[0.0, 1, "Ka2"].fix()
+    m.fs.leach.mscontactor.liquid_inherent_reaction_extent[0.0, :, "Ka2"].fix()
+
+
+def set_scaling(m):
+    """
+    Apply scaling factors to improve solver performance.
+    """
+    m.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+    for t in m.fs.time:
+        for s in m.fs.leach.mscontactor.elements:
+            for j in m.fs.coal.component_list:
+                if j not in ["Al2O3", "Fe2O3", "CaO", "inerts"]:
+                    m.scaling_factor[
+                        m.fs.leach.mscontactor.solid[t, s].mass_frac_comp[j]
+                    ] = 1e5
+                    m.scaling_factor[
+                        m.fs.leach.mscontactor.solid_inlet_state[t].mass_frac_comp[j]
+                    ] = 1e5
+                    m.scaling_factor[
+                        m.fs.leach.mscontactor.heterogeneous_reactions[
+                            t, s
+                        ].reaction_rate[j]
+                    ] = 1e5
+                    m.scaling_factor[
+                        m.fs.leach.mscontactor.solid[t, s].conversion_eq[j]
+                    ] = 1e3
+                    m.scaling_factor[
+                        m.fs.leach.mscontactor.solid_inlet_state[t].conversion_eq[j]
+                    ] = 1e3
+                    m.scaling_factor[
+                        m.fs.leach.mscontactor.heterogeneous_reactions[
+                            t, s
+                        ].reaction_rate_eq[j]
+                    ] = 1e5
+            for j in m.fs.leach_soln.component_list:
+                if j not in ["H2O"]:
+                    m.scaling_factor[
+                        m.fs.leach.mscontactor.liquid[t, s].conc_mass_comp[j]
+                    ] = 1e5
+                    m.scaling_factor[
+                        m.fs.leach.mscontactor.liquid_inlet_state[t].conc_mass_comp[j]
+                    ] = 1e5
```

Comment:
Instead of interacting with the suffix directly, please use `idaes.core.util.set_scaling_factor` to set these scaling factors.